src/hotspot/share/runtime/threads.cpp

```diff
@@ -548,7 +548,10 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
   // Initialize global modules
   jint status = init_globals();
   if (status != JNI_OK) {
-    main_thread->smr_delete();
+    if (!main_thread->has_pending_exception()) {
+      main_thread->smr_delete();
+    }
+    // Else don't delete main_thread as we need to report exceptions in the caller.
```

Comment:
Is it really okay to just leak main_thread?  If the process is going to die anyway then sure, but I'm not sure
that's true for all potential callers.  Of course, there's probably lots of other stuff getting leaked, so maybe
it doesn't matter.

Reply:
I see this was already discussed, with the same rationale for not caring about the leak.

Reply:
Thanks for looking at this Kim. With everything else we "leak" when we don't abort the process during VM init, adding the main thread to it would not make any difference IMO. But in any case with Coleen's suggestion we no longer "leak" the main thread as we will abort.